src/figuresClasses.ts

```diff
@@ -1,19 +1,97 @@
 export interface Figure {
-
+  shape: 'circle' | 'rectangle' | 'triangle';
+  color: 'red' | 'green' | 'blue';
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: 'triangle';
+
+  constructor(
+    public color: 'red' | 'green' | 'blue',
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    this.shape = 'triangle';
+    this.color = color;
+    this.a = a;
+    this.b = b;
+    this.c = c;
+
+    if (a + b <= c || a + c <= b || b + c <= a) {
+      throw new Error('Invalid triangle');
+    }
+  }
 
+  getArea(): number {
+    const p = (this.a + this.b + this.c) / 2;
```

Comment:
we can destructure this 
```suggestion
   const { a, b, c } = this;
    const p = (a + b + c) / 2;
```